Src/CSharpier/Printer/JoinClauseSyntax.cs

```diff
@@ -6,24 +6,26 @@ public partial class Printer
     {
         private Doc PrintJoinClauseSyntax(JoinClauseSyntax node)
         {
-            return Concat(
+            return Group(
                 this.PrintSyntaxToken(node.JoinKeyword, " "),
                 this.PrintSyntaxToken(node.Identifier, " "),
                 this.PrintSyntaxToken(node.InKeyword, " "),
                 this.Print(node.InExpression),
-                SpaceIfNoPreviousComment,
-                this.PrintSyntaxToken(node.OnKeyword, " "),
-                this.Print(node.LeftExpression),
-                SpaceIfNoPreviousComment,
-                this.PrintSyntaxToken(node.EqualsKeyword, " "),
-                this.Print(node.RightExpression),
-                node.Into != null
-                    ? Concat(
-                        SpaceIfNoPreviousComment,
-                        this.PrintSyntaxToken(node.Into.IntoKeyword, " "),
-                        this.PrintSyntaxToken(node.Into.Identifier)
-                    )
-                    : Doc.Null
+                Indent(
+                    Line,
+                    this.PrintSyntaxToken(node.OnKeyword, " "),
```

Comment:
This might be somewhat annoying, but I think this could would be much easier to parse for humans if you used `afterTokenIfNoTrailing: " "` as a named parameter. It's not really intuitive without looking at the definition.